app/services/billing_entities/update_invoice_payment_due_date_service.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module BillingEntities
+  class UpdateInvoicePaymentDueDateService < BaseService
+    def initialize(billing_entity:, net_payment_term:)
+      @billing_entity = billing_entity
+      @net_payment_term = net_payment_term
+      super
+    end
+
+    def call
+      ActiveRecord::Base.transaction do
+        # NOTE: Update payment_due_date if net_payment_term changed
+
+        if billing_entity.net_payment_term != net_payment_term
+          billing_entity.net_payment_term = net_payment_term
+          billing_entity.save!
```

Comment:
reading the old organization service I realized it assigns the new net payment term but does not save the record  changes (as you do here)... this looks like a bug to me. I'll raise it to the team.